ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -148,29 +191,78 @@ def _closest_waypoint_index(self):
         self.current_waypoint_ahead = prev_index
 
         return prev_index
-        
-    
+
+
     def publish_waypoints_ahead(self):
         """ Publishes a Lane of LOOKAHEAD_WPS waypoints /final_waypoint topic
         """
-        
         if self.base_waypoints is None or self.current_pose is None:
             return
 
         start_index = self._closest_waypoint_index()
         self.current_waypoint_ahead = start_index
 
         waypoint_indices = self._get_waypoint_indices(start_index, LOOKAHEAD_WPS)
-        
+
         lane = Lane()
         lane.header.frame_id = self.current_frame_id
         lane.header.stamp = rospy.Time.now()
         lane.header.seq = self.seq
         lane.waypoints = [self.base_waypoints[i] for i in waypoint_indices]
 
+        if self.closest_obstacle is None or self.current_tfl_state != 2:
+            # There is no traffic light near us, go full speed
+            self.current_velocity = self.current_velocity+0.5 if self.current_velocity < MAX_SPEED else self.current_velocity
+            speeds = np.linspace(self.current_velocity, MAX_SPEED, 1+(LOOKAHEAD_WPS//17))
+            full_speed = np.ones(16*LOOKAHEAD_WPS//17) * MAX_SPEED
+            speeds = 10# np.concatenate((speeds, full_speed))
+
+        else:
+            # There is a traffic light in front
+            if self.current_tfl_state ==  2:
+                remaining_wp = abs(self.closest_obstacle - self.current_waypoint_ahead)
+                distance = self.distance(lane.waypoints, 0, self.closest_obstacle - start_index)
+                """ Keeping comments for future reference in this commit only
+                rospy.logwarn("BreakSpeed")
+                rospy.logwarn("rem wp: " + str(remaining_wp))
+                if self.current_velocity > 1.5 and remaining_wp >= 55:
+                    rospy.logwarn("rm sp: " + str(100*(remaining_wp-LOOKAHEAD_WPS)/LOOKAHEAD_WPS))
+                    # Giving some extra space to reach speed == 0
+                    remaining_wp = remaining_wp - 15 if remaining_wp > 15 else remaining_wp
```

Comment:
these if-conditionals in one line are very confusing. I think the more obvious way is to just say:

if remaining_wp > 15:
  remaining_wp -= 15

this makes it much clearer what is going on.